### PR TITLE
StaticAnalyzer and ClangTidy PR test updates

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
please test

File: pr_testing/test_multiple_prs.sh [hunk 1]
```diff
@@ -687,11 +687,11 @@ if [ "X$DO_STATIC_CHECKS" = "Xtrue" -a "X$CMSSW_PR" != X -a "$RUN_TESTS" = "true
...
     scram b -k -j ${NCPU2} checker SCRAM_IGNORE_SUBDIRS=test 2>&1 | tee -a $WORKSPACE/llvm-analysis/runStaticChecks.log
+  grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep edm::eventsetup::EventSetupRecord::get | sort -u > $WORKSPACE/llvm-analysis/esrget-sa.txt
   cp -R $WORKSPACE/$CMSSW_IB/llvm-analysis/*/* $WORKSPACE/llvm-analysis || true
   if $IS_DEV_BRANCH && [ $(grep ': error: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | wc -l) -gt 0 ] ; then
     echo "EDM_ML_DEBUG_CHECKS;ERROR,Static Check build log,See Log,llvm-analysis/runStaticChecks.log" >> ${RESULTS_DIR}/static.txt
-  elif $IS_DEV_BRANCH && [ $(grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep '[threadsafety.ESRecordGetChecker]' | wc -l) -gt 0 ] ; then
-    grep ': warning: ' $WORKSPACE/llvm-analysis/runStaticChecks.log | grep edm::eventsetup::EventSetupRecord::get | sort -u > $WORKSPACE/llvm-analysis/esrget-sa.txt
+  elif $IS_DEV_BRANCH && [ $(wc -l $WORKSPACE/llvm-analysis/esrget-sa) -gt 0 ] ; then
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:694]: shouldn't  it be `esrget-sa.txt` instead of `esrget-sa`?

File: pr_testing/test_multiple_prs.sh [hunk 2]
```diff
@@ -703,12 +703,12 @@ if [ "X$DO_STATIC_CHECKS" = "Xtrue" -a "X$CMSSW_PR" != X -a "$RUN_TESTS" = "true
     echo '--------------------------------------'
     echo "Changed files:"
     echo ""
-    cat $WORKSPACE/changed-files
+    curl -s -L https://patch-diff.githubusercontent.com/raw/${REPOSITORY}/pull/${PULL_REQUEST}.patch | grep '^diff --git ' | sed 's|.* a/||;s|  *b/.*||' | sort | uniq > $WORKSPACE/all-changed-files.txt
```
Comment (smuzaffar) [inline pr_testing/test_multiple_prs.sh:706]: you need to use `$PR_REPO` and `$PR_NUM` in this script

Comment (gartung) [reply pr_testing/test_multiple_prs.sh:706]: PR_NUM is a md5 hash. PR_NUMBER appears to be the correct var.

File: pr_testing/test_multiple_prs.sh (hunk 1 above)
Comment (gartung) [reply pr_testing/test_multiple_prs.sh:694]: OK

**Comment by gartung** [comment]
please test with https://github.com/cms-sw/cmssw/pull/33215

**Comment by gartung** [comment]
please test with cms-sw/cmssw#33215

**Comment by gartung** [comment]
please test with cms-sw/cmssw#33215

**Comment by gartung** [comment]
abort

**Comment by gartung** [comment]
please test

**Pull request merged**